roles/generate_agent_iso/tasks/main.yml

```diff
@@ -0,0 +1,36 @@
+- name: Create podman auth dir
+  ansible.builtin.file:
+    path: "{{ ansible_env.HOME }}/.docker"
+    state: directory
+    mode: 0755
+    recurse: true
+
+- name: Copy pull_secret file.
+  ansible.builtin.copy:
+    content: "{{ pull_secret }}"
+    dest: "{{ ansible_env.HOME }}/.docker/config.json"
+    mode: 0644
+
+- name: Generate ISO
+  ansible.builtin.shell:
+    cmd: "{{ agent_based_installer_path }} --log-level=debug agent create image"
+    chdir: "{{ manifests_dir }}"
+  environment:
+    XDG_RUNTIME_DIR: "{{ config_file_path }}"
+    REGISTRY_AUTH_FILE: "{{ config_file_path }}/containers/auth.json"
```

Comment:
if we're already setting REGISTRY_AUTH_FILE is there any reason to copy to the default config.json location in the previous task ?

Reply:
Removed the `REGISTRY_AUTH_FILE` from the code base. Writting `{{ config_file_path }}/containers/auth.json` file but relying on `XDG_RUNTIME_DIR` to provide the file to the installer. 

To test removed `generated`, `~/.cache/agent` and `~/.docker` dirs from the host and ran an install with a disconnected node. Install completed without complaint.